unit/models/__init__.py

```diff
@@ -277,15 +277,19 @@ def from_json_api(data: Dict):
 
 
 class Merchant(UnitDTO):
-    def __init__(self, name: str, type: int, category: str, location: Optional[str]):
+    def __init__(self, name: str, _type: int, category: str, location: Optional[str]):
         self.name = name
-        self.type = type
+        self.type = _type
         self.category = category
         self.location = location
 
     @staticmethod
     def from_json_api(data: Dict):
-        return Merchant(data["name"], data["type"], data["category"], data.get("location"))
+        if data is None:
+            return None
+
+        return Merchant(data.get("name"), data.get("type"), data.get("category"), data.get("location"))
```

Comment:
not critical but you can leave it like it was with `data["name"], data["type"], data["category"]` - because if Merchant will return these fields are mandatory..

and just a syntax suggestion:
if data:
       return Merchant(...)
return None

Reply:
and let's cover this case in a test also:)